Sources/NIO/ByteBuffer-core.swift

```diff
@@ -231,7 +231,7 @@ public struct ByteBuffer {
         }
 
         private static func allocateAndPrepareRawMemory(bytes: Capacity, allocator: Allocator) -> UnsafeMutableRawPointer {
-            let bytes = Int(bytes)
+            let bytes = Int(bytes) /* Note(hh): this can fail on 32-bit (Int32 vs UInt32 Capacity)) */
```

Comment:
remove the Note ?

Reply:
I'm also not sure why this is an `Int`. `malloc` takes `size_t` which is a `UInt`

Reply:
ping @helje5

Reply:
Just checked: `size_t` is `unsigned int` on the 32-bit RasPi Ubuntu (I kinda assumed that it might be `signed int` because quite often you can't actually alloc more than 2GB on 32-bit platforms).

It probably did fail when one of the max-out tests passed in 0xDEADBEEF, which will fail the `Int(bytes)` conversion.

Why you are using `Int` I don't know, I didn't introduce this. But your code says:
```
let sysMalloc: @convention(c) (Int) -> UnsafeMutableRawPointer? = malloc
```
I don't know, maybe you should just use `size_t`?

Reply:
Moved this one to: #499.